bpfd/src/bpf.rs

```diff
@@ -913,31 +931,33 @@ impl BpfManager {
                 if let Some(map) = self.maps.get_mut(&m) {
                     map.used_by.push(id);
 
-                    // If map owner program still exists update it's maps_used_by_field
-                    if let Some(program) = self.programs.get_mut(&m) {
-                        program
-                            .data_mut()?
-                            .set_maps_used_by(Some(map.used_by.clone()));
+                    // Update all the programs using the same map with the updated map_used_by.
```

Comment:
I'm lost here for some reason....Say we have program A, B, and C Programs B/C are using the maps from Program A

When we add Program B all we have to do is update program A's maps_used_by right? 

So shouldn't this just be what I had before? 

Isn't this is now saying that programs A, B, AND C will all have the same "maps_used_by" field which makes no sense because only program A is the owner.

Reply:
Basically id map_owner_id is set I would expect maps_used_by to be NONE

Reply:
> When we add Program B all we have to do is update program A's maps_used_by right?
> So shouldn't this just be what I had before?

Yes, this is what we had before. And yes, `this is now saying that programs A, B, AND C will all have the same "maps_used_by" field`. With your implementation, remove Program A. How does the user know what programs are using Program A's maps now that it is gone? The user is flying blind at that point. Once we have the map object exposed to the user, this field can go away.

Reply:
I mean the user would just see "oh the map_owner_id doesn't actually exist anymore it must have been deleted" but I gotcha and am alright with it. Thanks for reminding me :)